src/semeio/forward_models/design_kw/design_kw.py

```diff
@@ -178,3 +184,41 @@ def rm_genkw_prefix(paramsdict, ignoreprefixes="LOG10_"):
         for keyvalue in keyvalues
         if keys.count(keyvalue[0]) == 1
     }
+
+
+def validate_configuration(
+    template_file_name: str, parameters_file_name: str = "parameters.txt"
+) -> None:
+    try:
+        if not os.path.exists(template_file_name):
+            raise ValueError(
+                f"Argument template_file: '{template_file_name}' is not an existing file"
+            )
+
+        if not os.path.exists(parameters_file_name):
+            raise ValueError(
+                f"Could not find required parameters file from DESIGN2PARAMS: '{parameters_file_name}'"
+            )
+        with open(parameters_file_name, encoding="utf-8") as parameters_file:
+            parameters = parameters_file.readlines()
+
+        key_vals = extract_key_value(parameters)
+
+        key_vals.update(rm_genkw_prefix(key_vals))
+
+        with open(template_file_name, encoding="utf-8") as template_file:
+            template = template_file.readlines()
+
+        for line in template:
+            if not is_comment(line):
+                for key, value in key_vals.items():
+                    line = line.replace(f"<{key}>", str(value))
+
+                all_matched(line, template_file_name, template)
```

Comment:
in `all_matched` this one does not raise `ValueError` but just logger.error. Not sure how to propagate it here, caplog?